src/featureManager.ts

```diff
@@ -75,14 +112,163 @@ export class FeatureManager {
         return !shortCircuitEvaluationResult;
     }
 
+    async #evaluateFeature(featureName: string, context: unknown): Promise<EvaluationResult> {
+        const featureFlag = await this.#provider.getFeatureFlag(featureName);
+        const result = new EvaluationResult(featureFlag);
+
+        if (featureFlag === undefined) {
+            return result;
+        }
+
+        // Ensure that the feature flag is in the correct format. Feature providers should validate the feature flags, but we do it here as a safeguard.
+        // TODO: move to the feature flag provider implementation.
+        validateFeatureFlagFormat(featureFlag);
+
+        // Evaluate if the feature is enabled.
+        result.enabled = await this.#isEnabled(featureFlag, context);
+
+        // Determine Variant
+        let variantDef: VariantDefinition | undefined;
+        let reason: VariantAssignmentReason = VariantAssignmentReason.None;
+
+        // featureFlag.variant not empty
+        if (featureFlag.variants !== undefined && featureFlag.variants.length > 0) {
+            if (!result.enabled) {
+                // not enabled, assign default if specified
+                if (featureFlag.allocation?.default_when_disabled !== undefined) {
+                    variantDef = featureFlag.variants.find(v => v.name == featureFlag.allocation?.default_when_disabled);
+                    reason = VariantAssignmentReason.DefaultWhenDisabled;
+                } else {
+                    // no default specified
+                    variantDef = undefined;
+                    reason = VariantAssignmentReason.DefaultWhenDisabled;
+                }
+            } else {
+                // enabled, assign based on allocation
+                if (context !== undefined && featureFlag.allocation !== undefined) {
+                    const variantAndReason = await this.#assignVariant(featureFlag, context as ITargetingContext);
+                    variantDef = variantAndReason.variant;
+                    reason = variantAndReason.reason;
+                }
+
+                // allocation failed, assign default if specified
+                if (variantDef === undefined && reason === VariantAssignmentReason.None) {
+                    if (featureFlag.allocation?.default_when_enabled !== undefined) {
+                        variantDef = featureFlag.variants.find(v => v.name == featureFlag.allocation?.default_when_enabled);
+                        reason = VariantAssignmentReason.DefaultWhenEnabled;

```

Comment:
No matter whether allocation.default_when_enabled is undefined or not, if the code enters here, the variant assignment reason should be DefualtWhenEnabled. Just like what you did for DefaultWhenDisabled

Reply:
good catch. fixed in 422b562